### PR TITLE
Asset Processor - Expand usage of source asset reference

### PR DESCRIPTION
**Pull request by amzn-mike**
## What does this PR do?

Update more Asset Processor code to work directly with SourceAssetReference type instead of passing around file paths.

Also added GetProductsBySourceNameScanFolderID query and updated several places to use this API instead when a specific source is intended.

This is part of the Asset Relocation work https://github.com/o3de/sig-content/issues/77

## How was this PR tested?

Ran existing unit tests.  Also ran AP.

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp [hunk 1]
```diff
@@ -413,6 +427,22 @@ namespace AzToolsFramework
                     SqlParam<const char*>(":sourcename"),
                     SqlParam<const char*>(":platform"));
 
+            static const char* QUERY_PRODUCT_BY_SOURCENAME_SCANFOLDERID_PLATFORM = "AzToolsFramework::AssetDatabase::QueryProductBySourceNameScanFolderIDPlatform";
+            static const char* QUERY_PRODUCT_BY_SOURCENAME_SCANFOLDERID_PLATFORM_STATEMENT =
+                "SELECT Products.*, Jobs.* FROM Sources LEFT JOIN Jobs "
+                                                                                "ON Sources.SourceID = Jobs.SourcePK INNER JOIN Products "
+                                                                                "ON Jobs.JobID = Products.JobPK WHERE "
+                                                                                "Jobs.Platform = :platform AND "
+                                                                                "Sources.SourceName = :sourcename AND Sources.ScanFolderPK = :scanfolderid;";
```
Comment (AMZN-stankowi) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp:436]: why are these lines tabbed so far over to the right?

File: Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp [hunk 2]
```diff
@@ -2053,14 +2053,14 @@ namespace AssetProcessor
         }
     }
 
-    void AssetProcessorManager::CheckMissingJobs(QString databasePathToFile, const ScanFolderInfo* scanFolder, const AZStd::vector<JobDetails>& jobsThisTime)
+    void AssetProcessorManager::CheckMissingJobs(const SourceAssetReference& sourceAsset, const ScanFolderInfo* scanFolder, const AZStd::vector<JobDetails>& jobsThisTime)
```
Comment (jackalbe) [inline Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp:2056]: "ScanFolderInfo* scanFolder" 
maybe this arg should be named "scanFolderInfo*" ?

File: Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp [hunk 3]
```diff
@@ -2161,7 +2160,7 @@ namespace AssetProcessor
         }
     }
 
-    void AssetProcessorManager::CheckModifiedSourceFile(QString normalizedPath, QString databaseSourceFile, const ScanFolderInfo* scanFolder)
+    void AssetProcessorManager::CheckModifiedSourceFile(const SourceAssetReference& sourceAsset, const ScanFolderInfo* scanFolder)
```
Comment (jackalbe) [inline Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp:2163]: "ScanFolderInfo* scanFolder"
maybe this arg should be named "scanFolderInfo*" ?

**Review by jackalbe — APPROVED** [review]

**Review by AMZN-stankowi — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp (hunk 1 above)
Comment (amzn-mike) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp:436]: Auto formatter which does whatever it wants... Ill update it.

File: Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp (hunk 2 above)
Comment (amzn-mike) [reply Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp]: Updated

File: Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp (hunk 3 above)
Comment (amzn-mike) [reply Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp]: Updated

**Pull request merged**